test/e2e_node/summary_test.go

```diff
@@ -0,0 +1,536 @@
+			Name: "test-empty-dir",
+			FsStats: stats.FsStats{
+				AvailableBytes: val(100 * gb),
+				CapacityBytes:  val(100 * gb),
+				UsedBytes:      val(1 * mb),
+			},
+		}},
+	}
+
+	upperBound = stats.Summary{
+		Node: stats.NodeStats{
+			SystemContainers: []stats.ContainerStats{
+				{
+					Name: "kubelet",
+					CPU: &stats.CPUStats{
+						UsageNanoCores:       val(2E9),
+						UsageCoreNanoSeconds: val(10E12),
+					},
+					Memory: &stats.MemoryStats{
+						AvailableBytes:  val(100 * gb),
+						UsageBytes:      val(10 * gb),
+						WorkingSetBytes: val(1 * gb),
+						RSSBytes:        val(1 * gb),
+						PageFaults:      val(1E9),
+						MajorPageFaults: val(100000),
+					},
+					Rootfs: &stats.FsStats{
+						AvailableBytes: val(100 * gb),
+						CapacityBytes:  val(100 * gb),
+						UsedBytes:      val(0), // Kubelet doesn't write.
+					},
+					Logs: &stats.FsStats{
+						AvailableBytes: val(100 * gb),
+						CapacityBytes:  val(100 * gb),
+						UsedBytes:      val(10 * gb),
+					},
+				},
+				{
+					Name: "runtime",
+					CPU: &stats.CPUStats{
+						UsageNanoCores:       val(2E9),
+						UsageCoreNanoSeconds: val(10E12),
+					},
+					Memory: &stats.MemoryStats{
+						AvailableBytes:  val(100 * gb),
+						UsageBytes:      val(10 * gb),
+						WorkingSetBytes: val(1 * gb),
+						RSSBytes:        val(1 * gb),
+						PageFaults:      val(1E9),
+						MajorPageFaults: val(100000),
+					},
+					Rootfs: &stats.FsStats{
+						AvailableBytes: val(100 * gb),
+						CapacityBytes:  val(100 * gb),
+						UsedBytes:      val(10 * gb),
+					},
+					Logs: &stats.FsStats{
+						AvailableBytes: val(100 * gb),
+						CapacityBytes:  val(100 * gb),
+						UsedBytes:      val(10 * gb),
+					},
+				},
+			},
+			CPU: &stats.CPUStats{
+				UsageNanoCores:       val(2E9),
+				UsageCoreNanoSeconds: val(10E12),
+			},
+			Memory: &stats.MemoryStats{
+				AvailableBytes:  val(100 * gb),
+				UsageBytes:      val(10 * gb),
+				WorkingSetBytes: val(1 * gb),
+				RSSBytes:        val(1 * gb),
+				PageFaults:      val(1E9),
+				MajorPageFaults: val(100000),
+			},
+			Network: &stats.NetworkStats{
+				RxBytes:  val(100 * gb),
+				RxErrors: val(100000),
+				TxBytes:  val(10 * gb),
+				TxErrors: val(100000),
+			},
+			Fs: &stats.FsStats{
+				AvailableBytes: val(100 * gb),
+				CapacityBytes:  val(100 * gb),
+				UsedBytes:      val(10 * gb),
+			},
+			Runtime: &stats.RuntimeStats{
+				ImageFs: &stats.FsStats{
+					AvailableBytes: val(100 * gb),
+					CapacityBytes:  val(100 * gb),
+					UsedBytes:      val(10 * gb),
+				},
+			},
+		},
+	}
+
+	ignoredFields = sets.NewString(
+		"Name",
+		"NodeName",
+		"PodRef",
+		"StartTime",
+		"UserDefinedMetrics",
+	)
+
+	allowedNils = sets.NewString(
+		".Node.SystemContainers[kubelet].Memory.AvailableBytes",
+		".Node.SystemContainers[runtime].Memory.AvailableBytes",
+		// TODO: Figure out why UsedBytes is nil on ubuntu-trusty-docker10 and coreos-stable20160622
```

Comment:
Given that we are attempting to support disk based eviction, this will be a serious issue. Can you file an issue?